src/metabase/actions/execution.clj

```diff
@@ -142,7 +162,42 @@
   (let [dashcard (api/check-404 (db/select-one DashboardCard
                                                :id dashcard-id
                                                :dashboard_id dashboard-id))
-        model-action (api/check-404 (db/select-one ModelAction :card_id (:card_id dashcard) :slug slug))]
+        model-action (api/check-404 (first (action/merged-model-action nil :card_id (:card_id dashcard) :slug slug)))]
     (if-let [action-id (:action_id model-action)]
       (execute-custom-action action-id request-parameters)
       (execute-implicit-action model-action request-parameters))))
+
+(defn- fetch-implicit-action-values
+  [dashboard-id model-action request-parameters]
+  (api/check (:requires_pk model-action) 400 (tru "Values can only be fetched for actions that require a Primary Key."))
+  (let [implicit-action (model-action->implicit-action model-action)
+        {:keys [prefetch-parameters]} (build-implicit-query model-action implicit-action request-parameters)
+        info {:executed-by api/*current-user-id*
+              :context :question
+              :dashboard-id dashboard-id}
+        card (db/select-one Card :id (:model_id model-action))
+        ;; prefilling a form with day old data would be bad
+        result (binding [persisted-info/*allow-persisted-substitution* false]
+                 (qp/process-query-and-save-execution!
+                   (qp.card/query-for-card card prefetch-parameters nil nil)
+                   info))
```

Comment:
I'm not sure about this bit that queries the model.
- I believe it should be saved to QueryExecution
- I'm not sure if there is a better way to build `info`
- I'm not sure if there's any other checks I need to make.

Reply:
We could check if `cache_ttl` is set on the card and if `(public-settings/enable-query-caching)` is truthy. Otherwise the results will not be cached, I think.